### PR TITLE
fix: ga4 v2 userproperties

### PR DESCRIPTION
**Pull request by utsabc**
## What are the changes introduced in this PR?

Added utility to remove nested objects from user_properties for ga4_v2

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3544/test-report.html

File: src/v0/destinations/ga4/utils.js [hunk 1]
```diff
@@ -554,21 +554,17 @@ const buildDeliverablePayload = (payload, Config) => {
...
-    if (
-      typeof propetyValue === 'string' ||
-      typeof propetyValue === 'number' ||
-      typeof propetyValue === 'boolean'
-    ) {
-      delete userProperties[key];
-      userProperties[key] = {
-        value: propetyValue,
-      };
+function sanitizeUserProperties(obj) {
```
Comment (koladilip) [inline src/v0/destinations/ga4/utils.js:557]: you don't want to mutate userProperties is that the change?

Comment (utsabc) [reply src/v0/destinations/ga4/utils.js:557]: No its basically I want to sanitize for any properties inside userProperties which are non primitive values

Comment (krishna2020) [inline src/v0/destinations/ga4/utils.js:557]:
```suggestion
function sanitizeUserProperties(userPropertyObject) {
```

Comment (utsabc) [reply src/v0/destinations/ga4/utils.js]: Addressed

**Review by koladilip — APPROVED** [review]

**Review by krishna2020 — APPROVED** [review]

**Pull request merged**